src/util/mod.rs

```diff
@@ -421,6 +426,28 @@ impl<T> Drop for MustConsumeVec<T> {
     }
 }
 
+/// Returns the tikv version information.
+pub fn tikv_version_info() -> String {
+    let fallback = "Unknown (env var does not exist when building)";
+    format!(
+        "\nRelease Version:   {}\
+         \nGit Commit Hash:   {}\
+         \nGit Commit Branch: {}\
+         \nUTC Build Time:    {}\
+         \nRust Version:      {}",
+        env!("CARGO_PKG_VERSION"),
+        option_env!("TIKV_BUILD_GIT_HASH").unwrap_or(fallback),
```

Comment:
Can we get these information at compile time?

Reply:
Please don't put it in `src/`, otherwise the TiKV library will be changed even if its source code is not changed, might cause cache invalidated. This will heavily increase compile time under different scenarios. See https://github.com/tikv/tikv/pull/3365 and https://github.com/tikv/tikv/pull/3362

Reply:
How about put it into `components`?

Reply:
That's ok and you can have a try. However you need to keep an eye on whether it is actually compiled after putting into `components` when its source code is not changed. If it is not re-compiled even after we changing `src/`, then it leads to wrong build info.